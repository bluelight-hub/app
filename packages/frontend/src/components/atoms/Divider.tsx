import React from 'react';
import { twMerge } from 'tailwind-merge';
import { BaseAtomProps } from '../../utils/types';

/**
 * Eine einfache Trennlinie f√ºr visuelle Separation von Inhalten
 */
type DividerProps = BaseAtomProps;

const Divider: React.FC<DividerProps> = ({
    className,
    'data-testid': dataTestId = 'divider'
}) => {
    return (
        <div
            data-testid={dataTestId}
            className={twMerge(`border-t border-gray-800/10 dark:border-gray-200/10 my-3`, className)}
        ></div>
    );
};

export default Divider; 