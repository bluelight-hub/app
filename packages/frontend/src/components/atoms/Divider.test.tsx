import { render, screen } from '@testing-library/react';
import { describe, expect, it } from 'vitest';
import Divider from './Divider';

/**
 * Tests für die Divider-Komponente
 */
describe('Divider', () => {
    // Unit Test - Komponente rendert ohne Fehler
    it('should render without crashing', () => {
        const { container } = render(<Divider data-testid="divider" />);
        const dividerElement = container.firstChild;
        expect(dividerElement).toBeInTheDocument();
    });

    // Unit Test - Klassennamen werden korrekt angewendet
    it('should apply default classes correctly', () => {
        const { container } = render(<Divider data-testid="divider" />);
        const dividerElement = container.firstChild;
        expect(dividerElement).toHaveClass('border-t');
        expect(dividerElement).toHaveClass('border-gray-800/10');
        expect(dividerElement).toHaveClass('dark:border-gray-200/10');
        expect(dividerElement).toHaveClass('my-3');
    });

    // Unit Test - zusätzliche Klassen können angegeben werden
    it('should merge additional className prop correctly', () => {
        const { container } = render(<Divider className="additional-class" data-testid="divider" />);
        const dividerElement = container.firstChild;
        expect(dividerElement).toHaveClass('additional-class');
        expect(dividerElement).toHaveClass('border-t');
    });

    // Integration Test - Verhalten im Layout
    it('should apply correct margin in a layout', () => {
        render(
            <div>
                <div>Content above</div>
                <Divider data-testid="divider" />
                <div>Content below</div>
            </div>
        );

        const dividerElement = screen.getByTestId('divider');
        expect(dividerElement).toHaveClass('my-3');
    });

    // Snapshot Test
    it('should match snapshot', () => {
        const { container } = render(<Divider data-testid="divider" />);
        expect(container).toMatchSnapshot();
    });

    // Snapshot Test mit benutzerdefinierten Klassen
    it('should match snapshot with custom className', () => {
        const { container } = render(<Divider className="custom-class" data-testid="divider" />);
        expect(container).toMatchSnapshot();
    });
}); 