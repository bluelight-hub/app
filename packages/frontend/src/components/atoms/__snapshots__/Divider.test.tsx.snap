// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Divider > should match snapshot 1`] = `
<div>
  <div
    class="border-t border-gray-800/10 dark:border-gray-200/10 my-3"
    data-testid="divider"
  />
</div>
`;

exports[`Divider > should match snapshot with custom className 1`] = `
<div>
  <div
    class="border-t border-gray-800/10 dark:border-gray-200/10 my-3 custom-class"
    data-testid="divider"
  />
</div>
`;
